hazelcast/src/main/java/com/hazelcast/map/impl/eviction/EvictionChecker.java

```diff
@@ -123,12 +124,17 @@ public double calculatePerNodeMaxRecordStoreSize(RecordStore recordStore) {
         int memberCount = nodeEngine.getClusterService().getSize(DATA_MEMBER_SELECTOR);
         int partitionCount = nodeEngine.getPartitionService().getPartitionCount();
 
-        final double perNodeMaxRecordStoreSize = (1D * configuredMaxSize * memberCount / partitionCount);
-        if (perNodeMaxRecordStoreSize < 1 && misconfiguredPerNodeMaxSizeWarningLogged.compareAndSet(false, true)) {
-            int minMaxSize = (int) Math.ceil((1D * partitionCount / memberCount));
-            logger.warning(format("The max size configuration for map \"%s\" does not allow any data in the map. "
-                    + "Given the current cluster size of %d members with %d partitions, max size should be at "
-                            + "least %d.", mapConfig.getName(), memberCount, partitionCount, minMaxSize));
+        double perNodeMaxRecordStoreSize = (1D * configuredMaxSize * memberCount / partitionCount);
+        if(perNodeMaxRecordStoreSize < 1) {
```

Comment:
Should compare against MIN_SANE_PER_PARTITION_SIZE here, rather than just 1, so that we never allow fewer records per node than the minimum.

Reply:
This line will also fail with CheckStyle (we require a space here `if (`).

Please use CheckStyle on your local machine: `mvn clean verify -Pcheckstyle -DskipTests`

Reply:
@rjatkins I left ability to set up cache size smaller than ```MIN_SANE_PER_PARTITION_SIZE``` just in case clients want to do that for whatever reason. I tried to fix the case, when cache size doesn't make sense (is definitely wrong). At the same time, partition size of ```1``` may make sense under some conditions. At the same time I can create ```useRecomendedMinSettings``` flag, which will force set partition size to  ```MIN_SANE_PER_PARTITION_SIZE``` (with another warn message probably). What do you think?

@Donnerbart thx for the feedback - I've fixed code formatting.